spec/rubocop/cop/rspec/prefer_before_over_setup_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::PreferBeforeOverSetup do
+  context 'when using `do ... end` block syntax' do
+    it 'accepts before call' do
+      expect_no_offenses(<<~RUBY)
+        before do
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+
+    it 'registers an offense on setup call' do
+      expect_offense(<<~RUBY)
+        setup do
+        ^^^^^^^^ Use `before` instead of `setup`.
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+  end
+
+  context 'when using curly braces block syntax' do
```

Comment:
As far as I'm aware, there's no difference between those two from AST standpoint. And it doesn't make sense to cover them both.

Reply:
I was thinking about ensuring that implementation stays the same but maybe it is redundant.